pkg/services/cloudmigration/cloudmigrationimpl/xorm_store.go

```diff
@@ -421,19 +421,22 @@ func (ss *sqlStore) UpdateSnapshotResources(ctx context.Context, snapshotUid str
 	})
 }
 
-func (ss *sqlStore) getSnapshotResources(ctx context.Context, snapshotUid string, page int, limit int) ([]cloudmigration.CloudMigrationResource, error) {
+func (ss *sqlStore) getSnapshotResources(ctx context.Context, snapshotUid string, params cloudmigration.SnapshotResultQueryParams) ([]cloudmigration.CloudMigrationResource, error) {
+	page, limit := params.ResultPage, params.ResultLimit
 	if page < 1 {
```

Comment:
I think similarly if we make the `page` and `lim` newtypes and validate it in the API layer before entering our domain we don't need this second validation here? Such validation can be done in constructing the newtype.